source/Octopus.Tentacle.Client/TentacleClient.cs

```diff
@@ -77,21 +80,34 @@ public async Task<UploadResult> UploadFile(string fileName, string path, DataStr
         {
             var operationMetricsBuilder = ClientOperationMetricsBuilder.Start();
 
+            UploadResult UploadFileAction(CancellationToken ct)
+            {
+                logger.Info($"Beginning upload of {fileName} to Tentacle");
+                var result = fileTransferServiceV1.UploadFile(path, package, new HalibutProxyRequestOptions(ct));
+                logger.Info("Upload complete");
+                return result;
+            }
+
             try
             {
-                return await rpcCallExecutor.ExecuteWithRetries(
-                    RpcCall.Create<IClientFileTransferService>(nameof(IClientFileTransferService.UploadFile)),
-                    ct =>
-                    {
-                        logger.Info($"Beginning upload of {fileName} to Tentacle");
-                        var result = fileTransferServiceV1.UploadFile(path, package, new HalibutProxyRequestOptions(ct));
-                        logger.Info("Upload complete");
-                        return result;
-                    },
-                    logger,
-                    abandonActionOnCancellation: false,
-                    operationMetricsBuilder,
-                    cancellationToken).ConfigureAwait(false);
+                if (rpcRetrySettings.RetriesEnabled)
+                {
+                    return await rpcCallExecutor.ExecuteWithRetries(
+                        RpcCall.Create<IClientFileTransferService>(nameof(IClientFileTransferService.UploadFile)),
+                        UploadFileAction,
+                        logger,
+                        abandonActionOnCancellation: false,
+                        operationMetricsBuilder,
+                        cancellationToken).ConfigureAwait(false);
+                }
+                else
+                {
+                    return rpcCallExecutor.Execute(
```

Comment:
Can we explicitly set `abandonActionOnCancellation: false,` to make it clear what the behaviour is